### PR TITLE
Fix fallback conditional for filter validation

### PR DESCRIPTION
**Pull request by ryanhamley**
## Launch Checklist

 - [x] briefly describe the changes in this PR
    - fixes https://github.com/mapbox/mapbox-gl-js/issues/11457 by fixing the conditional statement that allows for a fallback statement if `filter_${layerType}` is undefined
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog>Fix a bug that prevented some filters from being validated properly</changelog>`

### CODE REVIEW

**Review by mourner — COMMENTED** [review]
Can we add a test for this so that this doesn't regress again after a potential refactoring of validation code?

File: src/style-spec/validate/validate_filter.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ export default function validateFilter(options) {
         return validateExpression(extend({}, options, {
             expressionContext: 'filter',
             // We default to a layerType of `fill` because that points to a non-dynamic filter definition within the style-spec.
-            valueSpec: options.styleSpec[`filter_${layerType || 'fill'}`]
```
Comment (SnailBones) [inline src/style-spec/validate/validate_filter.js:16]: Good catch, much more readable!

I think that this change will have a conflict with the different approach @stepankuzmin is taking [here](https://github.com/mapbox/mapbox-gl-js/pull/11529/files#diff-270b52b46cd6599f7ffe53f4d49e9124f4a14c66d3e95b2f04cfd74dda1647bf).

**Review by SnailBones — APPROVED** [review]
Looks good to me!

Just one thought: your changes to `validate_filter.js` are going to trigger a merge conflict with https://github.com/mapbox/mapbox-gl-js/pull/11529, which also makes a change addressing the readability of that line. So perhaps consider reverting that change to simplify merging?

File: src/style-spec/validate/validate_filter.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ export default function validateFilter(options) {
...
+            valueSpec: options.styleSpec[`filter_${layerType}`] || options.styleSpec[`filter_fill`]
```
Comment (SnailBones) [inline src/style-spec/validate/validate_filter.js:16]: Nice improvement to readability! But I think that this change will have a conflict with the different approach @stepankuzmin is taking [here](https://github.com/mapbox/mapbox-gl-js/pull/11529/files#diff-270b52b46cd6599f7ffe53f4d49e9124f4a14c66d3e95b2f04cfd74dda1647bf).

Maybe consider reverting this to skip resolving the merge conflict?

```suggestion
            valueSpec: options.styleSpec[`filter_${layerType || 'fill'}`]
```

**Review by SnailBones — APPROVED** [review]

File: src/style-spec/validate/validate_filter.js (hunk 2 above)
Comment (ryanhamley) [reply src/style-spec/validate/validate_filter.js]: Thanks for pointing that out @SnailBones I reverted this change

**Pull request merged**